aptos-move/block-executor/src/executor.rs

```diff
@@ -256,7 +263,52 @@ where
                     maybe_layout,
                 ))?;
             }
-            // TODO(BlockSTMv2): delayed fields and aggregator v1.
+
+            // Apply aggregator v1 writes and deltas, using versioned data's V1 (write/add_delta) APIs.
+            // AggregatorV1 is not push-validated, but follows the same logic as delayed fields, i.e.
+            // commit-time validation in BlockSTMv2.
+            for (key, value) in output.aggregator_v1_write_set().into_iter() {
+                prev_modified_aggregator_v1_keys.remove(&key);
+
+                versioned_cache.data().write(
+                    key,
+                    idx_to_execute,
+                    incarnation,
+                    Arc::new(value),
+                    None,
+                );
+            }
+            for (key, delta) in output.aggregator_v1_delta_set().into_iter() {
+                prev_modified_aggregator_v1_keys.remove(&key);
+                versioned_cache.data().add_delta(key, idx_to_execute, delta);
+            }
+
+            for (id, change) in output.delayed_field_change_set().into_iter() {
```

Comment:
This is the same as in V1, can we factor this out in a method and reuse across `execute` and `execute_v2`?